### DISCUSSION TITLE
Linters status in general and unusedwrite check in particular

### DISCUSSION THREAD

**Discussion by ruz** [golangci/golangci-lint#5108] [Q&A] [answered]
Hello,

Started using vscode clone. Just by incident noticed highlighted code in the IDE.

This was [unusedwrite Analyzer](https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/unusedwrite#hdr-Analyzer_unusedwrite) error. Looks like `gopls check <filename>` runs this analyzer.

Couldn't find any other ways to run it. Tried golangci-lint, staticcheck and go vet. 

I would prefer to have one linters runner in the project. How do I integrate this analyzer into golangci-lint?

This analyzer finds common annoying bug in for loops where people change fields of the iterator forgetting to save the change. This mistake is so easy to make:

```go
for _, e := range aList {
    e.x = "foo"
}
```

**Comment by ldez** [comment, marked as answer]
Hello,

You should enable the check:

```yml
linters-settings:
  govet:
    enable:
      - unusedwrite
```

https://golangci-lint.run/usage/linters/#govet

**Reply by ruz** [reply]
Thanks. This page is so big. Either slow internet or whatever, but my initial ctrl+f was unsuccessful.

Any idea why this check is disabled by default?

**Reply by ldez** [reply]
golangci-lint uses the same default as go vet.

https://github.com/golang/go/blob/b56645a87b28840a180d64077877cb46570b4176/src/cmd/vet/main.go#L49-L81
